Tests/ApolloCodegenTests/AnimalKingdomAPI/AnimalKingdomIRCreationTests.swift

```diff
@@ -1038,4 +1038,54 @@ final class AnimalKingdomIRCreationTests: XCTestCase {
     expect(selectionSet.parentType).to(equal(GraphQLObjectType.mock("Height")))
     expect(actual).to(shallowlyMatch(self.expected))
   }
+
+  // MARK: - Referenced Fragment Tests
+
+  func test__referencedFragments__AllAnimalsQuery_isCorrect() throws {
+    // given
+    let operation = compilationResult.operations.first { $0.name == "AllAnimalsQuery" }
+
+    // when
+    let expected: [CompilationResult.FragmentDefinition] = [
+      "HeightInMeters",
+      "WarmBloodedDetails",
+      "PetDetails"
+    ].map { expectedName in
+      compilationResult.fragments.first(where:{ $0.name == expectedName })!
+    }
+
+    // then
+    expect(operation?.referencedFragments).to(equal(expected))
+  }
+
+  func test__referencedFragments__HeightInMeters_isCorrect() throws {
+    // given
+    let fragment = compilationResult.fragments.first { $0.name == "HeightInMeters" }
+
+    // then
+    expect(fragment?.referencedFragments).to(beEmpty())
+  }
+
+  func test__referencedFragments__WarmBloodedDetails_isCorrect() throws {
+    // given
+    let fragment = compilationResult.fragments.first { $0.name == "WarmBloodedDetails" }
+
+    // when
+    let expected: [CompilationResult.FragmentDefinition] = [
+      "HeightInMeters"
+    ].map { expectedName in
+      compilationResult.fragments.first(where:{ $0.name == expectedName })!
+    }
+
+    // then
+    expect(fragment?.referencedFragments).to(equal(expected))
+  }
+
+  func test__referencedFragments__PetDetails_isCorrect() throws {
```

Comment:
These are kind of odd tests because they test the example animal kingdom API. When that API, these tests need to be changed to accurately represent the correct information there. So changing the name also just doesn't add value. The only reason "shouldNotHaveReferencedFragments" is true is because the Pet Details fragment doesn't have any right now. It's not truly a test that "a fragment with no referenced fragments should have an empty referenced fragments array." The unit tests for this PR are the tests in the JS code. These are just some added integration tests.